components/org.wso2.carbon.um.ws.service/src/main/resources/META-INF/services.xml

```diff
@@ -35,6 +35,210 @@
         </transports>
         <parameter name="ServiceClass" locked="false">org.wso2.carbon.um.ws.service.UserStoreManagerService</parameter>
         <parameter name="AuthorizationAction" locked="false">/permission/admin/manage/identity</parameter>
+
+        <operation name="addRole">
+            <parameter name="AuthorizationAction"
+                       locked="true">/permission/admin/manage/identity/rolemgt/create
+            </parameter>
+        </operation>
+
+        <operation name="addUser">
+            <parameter name="AuthorizationAction"
+                       locked="true">/permission/admin/manage/identity/usermgt/create
+            </parameter>
+        </operation>
+
+        <operation name="addUserClaimValue">
+            <parameter name="AuthorizationAction"
+                       locked="true">/permission/admin/manage/identity/usermgt/update
+            </parameter>
+        </operation>
+
+        <operation name="addUserClaimValues">
+            <parameter name="AuthorizationAction"
+                       locked="true">/permission/admin/manage/identity/usermgt/update
+            </parameter>
+        </operation>
+
+        <operation name="authenticate">
+            <parameter name="AuthorizationAction"
+                       locked="true">/permission/admin/manage/identity
```

Comment:
Should we make this specific as well? When we provide a top-level permission, that user inherits all the permissions that are in further down the tree such as /permission/admin/manage/identity/usermgt/update.

Reply:
Agreed in mail thread